src/main/java/com/sourcery/gymapp/backend/userProfile/mapper/UserProfileMapper.java

```diff
@@ -0,0 +1,50 @@
+package com.sourcery.gymapp.backend.userProfile.mapper;
+
+import com.sourcery.gymapp.backend.globalConfig.AuditorConfig;
+import com.sourcery.gymapp.backend.userProfile.dto.UserProfileDto;
+import com.sourcery.gymapp.backend.userProfile.exception.UserNotFoundException;
+import com.sourcery.gymapp.backend.userProfile.model.UserProfile;
+import com.sourcery.gymapp.backend.userProfile.repository.UserProfileRepository;
+import lombok.RequiredArgsConstructor;
+import java.util.UUID;
+
+import org.springframework.data.domain.AuditorAware;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class UserProfileMapper {
+
+    private final AuditorAware<UUID> auditorAware;
+    private final UserProfileRepository userProfileRepository;
```

Comment:
Mapper should not have access to repository or auditor, it should be on service layer